vault/login_mfa.go

```diff
@@ -106,8 +106,14 @@ func (b *SystemBackend) loginMFAPaths() []*framework.Path {
 	}
 }
 
-func genericOptionalUUIDRegex(name string) string {
-	return fmt.Sprintf("(/(?P<%s>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}))?", name)
+// uuidRegex crafts a regex for use in URL paths, somewhat similar to framework.GenericNameRegex, but only accepting
+// UUIDs, and only lowercase UUIDs at that.
```

Comment:
just curious, why only lowercase uuid's? is that a restriction of the mfa api's?

Reply:
Indeed, it's a pre-existing implementation choice of the identity/mfa/method/ APIs, that I'm writing down to warn people against re-using the code elsewhere, where that choice may not apply.